src/control/server/ctl_storage_rpc.go

```diff
@@ -258,3 +259,42 @@ func (c *ControlService) StorageFormat(ctx context.Context, req *ctlpb.StorageFo
 
 	return resp, nil
 }
+
+// StorageNvmeUnbind unbinds SSD from kernel and binds to user-space to allow DAOS to use it.
+func (c *ControlService) StorageNvmeUnbind(ctx context.Context, req *ctlpb.NvmeUnbindReq) (*ctlpb.NvmeUnbindResp, error) {
+	c.log.Debugf("received StorageNvmeUnbind RPC %v", req)
+
+	if req == nil {
+		return nil, errors.New("nil request")
+	}
+
+	cu, err := user.Current()
+	if err != nil {
+		return nil, errors.Wrap(err, "get username")
+	}
+
+	prepReq := storage.BdevPrepareRequest{
+		// zero as hugepages already allocated on start-up
+		HugePageCount: 0,
+		TargetUser:    cu.Username,
+		PCIAllowList:  req.PciAddr,
+		Reset_:        false,
+	}
+
+	resp := new(ctlpb.NvmeUnbindResp)
+	if _, err := c.NvmePrepare(prepReq); err != nil {
```

Comment:
agreed, I will make the change if I need to re-push

Reply:
not currently because it's used directly by cmd/daos_server, this should be delegated to the admin binary though so when that is fixed we can remove the export